#!/usr/bin/env python3
# imports in try-block below, to help check for env
def print_no_env_error():
    print(" -> Your env for this project is not active.")
    print(" -> Please run 'source <path to your venv>/bin/active' and try again.")
    return

# ---- IMPORTS ----
try:
    import sys, os, pytest
except ImportError as e:
    raise ImportError("Import(s) not found.\n{0}".format(print_no_env_error()))
# -----------------

def verify_env_running():
    if 'VIRTUAL_ENV' not in os.environ:
        print_no_env_error()
        return False
    return True

def run_test_suite():
    # Run the test suite, and exit with whatever exit code the testsuite gives
    #   if all tests pass, it exits with 0 and is allowed to push

    test_dir = os.path.normpath(os.path.join(os.path.dirname(__file__), "..", "..", "test_cameron"))
    sys.exit( pytest.main(["test_cameron"]) )


if __name__ == "__main__":
    if not verify_env_running():
        sys.exit(6)
    run_test_suite()
    sys.exit(0)


# Exit code 0:    All tests were collected and passed successfully
# Exit code 1:    Tests were collected and run but some of the tests failed
# Exit code 2:    Test execution was interrupted by the user
# Exit code 3:    Internal error happened while executing tests
# Exit code 4:    pytest command line usage error
# Exit code 5:    No tests were collected
# Exit code 6:  Your virtual env is not running
